### PR TITLE
[TC-0806] BE: recommitment

### PR DESCRIPTION
**Pull request by chelsea-EYDS**
[TC-0806](https://bcdevex.atlassian.net/browse/TC-0806)

Objective:

- recommitment entity
- recommitment cycle entity
- (Test on branch TC-809)

### CODE REVIEW

File: backend/src/database/migrations/1731988334892-ministry-travelmigration.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import { MigrationInterface, QueryRunner } from 'typeorm';
```
Comment (fw-noel) [inline backend/src/database/migrations/1731988334892-ministry-travelmigration.ts:1]: Is this an old migration?

File: backend/src/personnel/personnel.service.ts [hunk 2]
```diff
@@ -453,6 +457,7 @@ export class PersonnelService {
       .createQueryBuilder('personnel')
       .leftJoinAndSelect('personnel.workLocation', 'workLocation')
       .leftJoinAndSelect('personnel.homeLocation', 'homeLocation')
+      .leftJoinAndSelect('personnel.recommitment', 'recommitment')
```
Comment (fw-noel) [inline backend/src/personnel/personnel.service.ts:460]: I don't love that we're getting this all year round either. Not a dealbreaker by any means though. But if you have any thoughts, feel free to change.

Comment (chelsea-EYDS) [reply backend/src/personnel/personnel.service.ts:460]: We may need this if we plan to display the recommitment status throughout the year (it sounds like they want this included on the member profile for people who did not recommit so they know the reason for being in active)

File: backend/src/database/entities/personnel.entity.ts [hunk 3]
```diff
@@ -127,9 +128,13 @@ export class PersonnelEntity extends BaseEntity {
   })
   ministry: Ministry;
 
-  @Column({name: 'division', type: 'varchar', length: 100, nullable: true})
+  @Column({ name: 'division', type: 'varchar', length: 100, nullable: true })
   division?: string;
 
+  @OneToOne(() => RecommitmentEntity, (r) => r.memberId, { nullable: true })
+  @JoinColumn({ name: 'recommitment', referencedColumnName: 'memberId' })
```
[the remark this answers is not in the document]
Comment (chelsea-EYDS) [reply backend/src/database/entities/personnel.entity.ts:135]: memberId

**Review by fw-noel — DISMISSED** [review]

**Pull request merged**